xdsl/irdl/declarative_assembly_format.py

```diff
@@ -801,6 +802,21 @@ class OptionalAttributeVariable(AttributeVariable, OptionalVariable):
     """
 
 
+class OptionalUnitAttrVariable(OptionalAttributeVariable):
+    """
+    An optional UnitAttr variable
```

Comment:
Could you please expand a bit here why this is needed? Something about UnitAttr not being printed in the IR, so special handling is needed, maybe with a link to the MLIR documentation you posted in the other comment.

Reply:
Doing as suggested